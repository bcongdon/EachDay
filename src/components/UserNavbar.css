.navbar-logo-link {
    color: #000;
}

.navbar-logo-link:hover {
    color: #333;
}