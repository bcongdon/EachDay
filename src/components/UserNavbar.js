import React, { Component } from 'react'
import { Menu, Dropdown } from 'semantic-ui-react'
import Gravatar from './Gravatar'
import { connect } from 'react-redux'
import { logoutUser, pushPage } from '../actions'
import { PropTypes } from 'prop-types'
import Logo from './Logo'
import { Link } from 'react-router-dom'
import './UserNavbar.css'

class UserNavbar extends Component {
  constructor(props) {
    super(props)
    this.logoutClick = this.logoutClick.bind(this)
    this.profileClick = this.profileClick.bind(this)
    this.dashboardClick = this.dashboardClick.bind(this)
  }

  getTrigger() {
    return (
      <span>
        <Gravatar email={this.props.user.email} /> {this.props.user.name}
      </span>
    )
  }

  logoutClick(event, data) {
    this.props.logoutUser()
  }

  profileClick(event, data) {
    this.props.pushPage('/profile')
  }

  dashboardClick(event, data) {
    this.props.pushPage('/dashboard')
  }

  render() {
    return (
      <Menu size='large' borderless>
        <Menu.Item>
          <Link className='navbar-logo-link' to='/dashboard'><Logo /></Link>
        </Menu.Item>
        <Menu.Menu position='right'>
          <Menu.Item>
            <Dropdown trigger={this.getTrigger()} pointing='top right' icon={null}>
              <Dropdown.Menu>
                <Dropdown.Item text='Journal' icon='calendar outline' onClick={this.dashboardClick} />
                <Dropdown.Item text='Account' icon='user' onClick={this.profileClick} />
                <Dropdown.Item text='Sign Out' icon='sign out' onClick={this.logoutClick} />
              </Dropdown.Menu>
            </Dropdown>
          </Menu.Item>
        </Menu.Menu>
      </Menu>
    )
  }
}

UserNavbar.propTypes = {
  user: PropTypes.object.isRequired,
  logoutUser: PropTypes.func.isRequired,
  pushPage: PropTypes.func.isRequired
}

function mapStateToProps(state) {
  return {
    user: state.auth.user
  }
}

export default connect(mapStateToProps, { logoutUser, pushPage })(UserNavbar)
