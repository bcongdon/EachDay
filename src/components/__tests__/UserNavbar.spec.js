import React from 'react'
import { UserNavbar } from '../UserNavbar'
import { mount, shallow } from 'enzyme'
import { Menu } from 'semantic-ui-react'
import toJson from 'enzyme-to-json'
jest.mock('universal-cookie')

const user = {
  name: 'foo',
  email: 'bar@baz.com'
}

it('renders correctly', () => {
  const pushPage = jest.fn()
  const logoutUser = jest.fn()
  const navbar = shallow(
    <UserNavbar
      user={user}
      location={{pathname: '/dashboard'}}
      pushPage={pushPage}
      logoutUser={logoutUser}
      />
  )
  expect(toJson(navbar)).toMatchSnapshot()
})

it('should reload dashboard window on a simulated refresh', () => {
  const pushPage = jest.fn()
  const logoutUser = jest.fn()
  const navbar = mount(
    <UserNavbar
      user={user}
      location={{pathname: '/dashboard'}}
      pushPage={pushPage}
      logoutUser={logoutUser}
      />
  )
  global.location.reload = jest.fn()
  navbar.find(Menu.Item).first().simulate('click')
  expect(global.location.reload).toBeCalled()
  expect(pushPage).not.toBeCalled()
})

it('should push page on a non-dashboard click', () => {
  const pushPage = jest.fn()
  const logoutUser = jest.fn()
  const navbar = mount(
    <UserNavbar
      user={user}
      location={{pathname: '/'}}
      pushPage={pushPage}
      logoutUser={logoutUser}
      />
  )
  global.location.reload = jest.fn()
  navbar.find(Menu.Item).first().simulate('click')
  expect(global.location.reload).not.toBeCalled()
  expect(pushPage).toBeCalled()
})
