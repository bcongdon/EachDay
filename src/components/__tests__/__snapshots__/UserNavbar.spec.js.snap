// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Menu
  borderless={true}
  size="large"
>
  <MenuItem
    onClick={[Function]}
  >
    <Logo />
  </MenuItem>
  <MenuMenu
    position="right"
  >
    <MenuItem>
      <Dropdown
        additionLabel="Add "
        additionPosition="top"
        closeOnBlur={true}
        icon={null}
        noResultsMessage="No results found."
        openOnFocus={true}
        pointing="top right"
        renderLabel={[Function]}
        selectOnBlur={true}
        trigger={
          <span>
            <Gravatar
              avatar={true}
              email="bar@baz.com"
              size={80}
            />
             
            foo
          </span>
        }
      >
        <DropdownMenu>
          <DropdownItem
            icon="calendar outline"
            onClick={[Function]}
            text="Journal"
          />
          <DropdownItem
            icon="user"
            onClick={[Function]}
            text="Account"
          />
          <DropdownItem
            icon="sign out"
            onClick={[Function]}
            text="Sign Out"
          />
        </DropdownMenu>
      </Dropdown>
    </MenuItem>
  </MenuMenu>
</Menu>
`;
